src/bp/ui-studio/src/web/views/OneFlow/diagram/Forms/SubWorkflowForm/InputForm.tsx

```diff
@@ -26,18 +26,35 @@ const InputForm: FC<Props> = ({
   const [fieldTypes, setFieldTypes] = useState({})
 
   useEffect(() => {
-    setFieldTypes(_.mapValues(formData, item => (item?.source === 'variable' ? 'variable' : 'text')))
+    setFieldTypes(
+      _.mapValues(formData, (item, variable) => {
+        const details = subFlowVars.find(x => x.params.name === variable)
+        const inputType = variables.primitive.find(x => x.id === details?.type)?.config.inputType
+
+        return item?.source === 'variable' ? 'variable' : inputType
+      })
+    )
   }, [customKey])
 
-  const fields = subFlowVars.map<FormField>(({ params, type }) => ({
-    type: fieldTypes[params.name] ?? 'variable',
-    key: params.name,
-    label: params.name,
-    variableTypes: [type],
-    placeholder: 'module.builtin.enterValue',
-    onClick: field => setFieldTypes({ ...fieldTypes, [field.key]: field.type === 'variable' ? 'text' : 'variable' }),
-    defaultVariableType: type
-  }))
+  const fields = subFlowVars.map<FormField>(({ params, type }) => {
+    const inputType = variables.primitive.find(x => x.id === type)?.config.inputType
+    const realType = params.elements?.length ? 'select' : inputType
```

Comment:
what's realtype ?

Reply:
Just a placeholder name, because there's the field input type, but if there's a list of possible options, then it becomes a select